af/shovel/centrifugation.py

```diff
@@ -326,163 +329,491 @@ def __flush_stubborn(self):
     def close(self):
         self.flush()
         self.__buf.close()
+    @property
+    def closed(self):
+        return self.__buf.closed
 
-def get_checked_code_ver(pgconn, autoclaved_index, bucket):
-    # Returns:
-    #   None -- the bucket was never loaded
-    #   int  -- code version stamped in `autoclaved` table for these files
-    #
-    # Check that postgresql and autoclaved/${bucket}/index.json.gz ...
-    # - have same list of data files, names and SHA1 are equal
-    # - have same count of reports (type: report) and measurements (type: datum)
-    #
-    # Number of distinct frames in PG and {type: frame} may differ as there are frames without measurements:
-    # $ zcat 2017-03-21/index.json.gz | awk '/"type": ?"frame"/ {a = 0; s = ""} {s = s $0} /"type": ?"datum"/ { a +=1 } (/"type": ?"\/frame"/ && a == 0) {print s; s = ""}' | shuf -n 5
-    # {"file_off": 14628392, "file_size": 15, "text_off": 60180480, "text_size": 0, "type": "frame"}{"type": "/frame"}
-    # {"file_off": 14508951, "file_size": 74, "text_off": 59792468, "text_size": 9132, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 11295485, "file_size": 60, "text_off": 43934392, "text_size": 5448, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 9654710, "file_size": 28, "text_off": 41709591, "text_size": 1, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    # {"file_off": 26744767, "file_size": 28, "text_off": 116908344, "text_size": 1, "type": "frame"}{"type": "/report"}{"type": "/frame"}
-    #
-    # TODO: {type: badblob} is not accounted as it's not currently imported during centrifugation stage.
-
-    files = set()
-    use_report = True
-    report_count, datum_count = 0, 0
+def load_autoclaved_index(autoclaved_index):
+    # Returns: {filename: file_sha1 for autoclaved}
+    files = {}
     with gzip.GzipFile(autoclaved_index, 'r') as indexfd:
         for _, doc in autoclaving.stream_json_blobs(indexfd):
+            if 'file"' not in doc:
+                continue
             doc = ujson.loads(doc)
-            t = doc['type']
-            if t == 'file':
+            if doc['type'] == 'file':
                 filename = doc['filename']
-            elif t == '/file':
-                files.add((b64decode(doc['file_sha1']), filename))
-            elif t == 'report':
-                if doc['textname'] not in DUPLICATE_REPORTS:
-                    report_count += 1
-                else:
-                    use_report = False
-            elif t == '/report':
-                use_report = True
-            elif t == 'datum' and use_report:
-                datum_count += 1
-
+            elif doc['type'] == '/file':
+                files[filename] = b64decode(doc['file_sha1'])
+    return files
+
+def load_autoclaved_db_todo(pgconn, files, bucket):
+    # Returns: {files to ingest}, {files to re-ingest}, {files to re-process}
```

Comment:
Maybe it's wise to mention that the returned type is a `dict_keys()` instance.

Reply:
Fair enough. I would still prefer to explicitly state it as I was unsure in the beginning if you were using that syntax to say they are lists or sets.